api/cpp/include/slint.h

```diff
@@ -89,16 +89,13 @@ class WindowAdapterRc
 {
 public:
     explicit WindowAdapterRc(cbindgen_private::WindowAdapterRcOpaque adopted_inner)
-        : inner(adopted_inner)
     {
+        assert_main_thread();
+        cbindgen_private::slint_windowrc_clone(&adopted_inner, &inner);
     }
     WindowAdapterRc() { cbindgen_private::slint_windowrc_init(&inner); }
     ~WindowAdapterRc() { cbindgen_private::slint_windowrc_drop(&inner); }
-    WindowAdapterRc(const WindowAdapterRc &other)
-    {
-        assert_main_thread();
-        cbindgen_private::slint_windowrc_clone(&other.inner, &inner);
-    }
+    WindowAdapterRc(const WindowAdapterRc &other) : WindowAdapterRc(other.inner) { }
```

Comment:
The forwarded constructor does the clone

Reply:
Then I think the constructor's parameter should not be called `adopted_inner`, as it isn't adopting anymore.

That said, if we (accidentally) call this copy constructor now, then it will cause a crash if the `other` dies before the newly constructed one, won't it?